parquet-thrift/src/main/java/org/apache/parquet/thrift/struct/CompatibilityChecker.java

```diff
@@ -43,24 +46,81 @@ public CompatibilityReport checkCompatibility(ThriftType.StructType oldStruct, T
 
 class CompatibilityReport {
   boolean isCompatible = true;
+  boolean hasEmptyStruct = false;
   List<String> messages = new ArrayList<String>();
 
   public boolean isCompatible() {
     return isCompatible;
   }
 
+  public boolean hasEmptyStruct() {
+    return hasEmptyStruct;
+  }
+
   public void fail(String message) {
     messages.add(message);
     isCompatible = false;
   }
 
+  public void emptyStruct(String message) {
+    messages.add(message);
+    hasEmptyStruct = true;
+  }
+
   public List<String> getMessages() {
     return messages;
   }
+
+  public String prettyMessages() {
+    StringBuffer message = new StringBuffer();
+    for(String m: messages) {
+      message.append(m);
+      message.append("\n");
+    }
+    return message.toString();
+  }
+
+  @Override
+  public String toString() {
+    return "CompatibilityReport{" +
+        "isCompatible=" + isCompatible +
+        ", hasEmptyStruct=" + hasEmptyStruct +
+        ", messages=\n" + prettyMessages() +
+        '}';
+  }
 }
 
 class CompatibleCheckerVisitor implements ThriftType.TypeVisitor {
+
+  /**
+   * log the path when a incompatible field is found, so it's easier to track down the incompatible field
+   */
+  private static class FieldPath {
```

Comment:
can we use the FieldPath that we already have? (the one in my PR would work I think)